### PR TITLE
add support for the asymmetric Encrypt and Decrypt restful API

### PR DESCRIPTION
**Pull request by Liang1XZhao**
this restful API will exposed to public as:
- POST <ehsm_srv_address>/ehsm?Action=AsymmetricEncrypt
- POST <ehsm_srv_address>/ehsm?Action=AsymmetricDecrypt
add the corresponding test cases.

change server port 3000 to 9000.
Napi_* Method from ehsm_kms_server.JS extract to ehsm_napi.js

test: passed the unittest

Signed-off-by: Liang1XZhao <liang1x.zhao@intel.com>

### CODE REVIEW

File: ehsm_kms_service/ehsm_kms_params.js [hunk 1]
```diff
@@ -106,7 +106,29 @@ const ehsm_kms_params = {
...
-  }
+  },
+  AsymmetricEncrypt: {
+    cmk_base64: {
+      type: 'string',
+      minLength: 1
+    },
+    plaintext: {
+      type: 'string',
+      maxLength: _6kb_length,
```
Comment (yang8621) [inline ehsm_kms_service/ehsm_kms_params.js:117]: only support 318 Bytes for RSA encryption

File: ehsm_kms_service/ehsm_kms_params.js [hunk 2]
```diff
@@ -106,7 +106,29 @@ const ehsm_kms_params = {
...
+    },
+  },
+  AsymmetricDecrypt: {
+    cmk_base64: {
+      type: 'string',
+      minLength: 1
+    },
+    ciphertext_base64: {
+      type: 'string',
+      maxLength: _6kb_length,
```
Comment (yang8621) [inline ehsm_kms_service/ehsm_kms_params.js:128]: The Length is not 6KB,  I need to check.

File: core/App/ehsm_core_test.cpp [hunk 3]
```diff
@@ -126,144 +126,52 @@ void test_AES128()
...
-    }
-    memset(digest.data, 'B', digest.datalen);
-    dump_data(digest.data, digest.datalen);
-
-    signature.datalen = 0;
-    ret = Sign(&cmk, &digest, &signature);
-    if (ret != EH_OK) {
-        printf("Failed(%d) to get data size of Sign!\n", ret);
+    returnJsonChar = NAPI_CreateKey(EH_RSA_3072, EH_INTERNAL_KEY);
+    if(returnJsonChar == nullptr){
```
[the remark this answers is not in the document]
Comment (Liang1XZhao) [reply core/App/ehsm_core_test.cpp:140]: will be fix

File: ehsm_kms_service/ehsm_kms_server.js [hunk 4]
```diff
@@ -352,18 +164,30 @@ app.post('/ehsm', function (req, res) {
...
-      const { cmk_base64, digest, signature_base64 } = PAYLOAD;
-      napi_result(ACTION ,res, [cmk_base64, digest, signature_base64]);
-  }else {
+  /**
+   * Verify
+   */
+    const { cmk_base64, digest, signature_base64 } = PAYLOAD;
+    napi_result(ACTION ,res, [cmk_base64, digest, signature_base64]);
+  } else if(ACTION === apis.AsymmetricEncrypt) {
+  /**
```
Comment (yang8621) [inline ehsm_kms_service/ehsm_kms_server.js:179]: Will js layer check the input validity, e.g. the maxlength, minLength？

File: core/App/ehsm_core_test.cpp [hunk 5]
```diff
@@ -126,144 +126,52 @@ void test_AES128()
...
+    returnJsonChar = NAPI_AsymmetricDecrypt(cmk_base64, ciphertext_base64);
+    if(returnJsonChar == nullptr){
+        printf("NAPI_AsymmetricDecrypt failed!\n");
         goto cleanup;
     }
-    printf("Verify signature SUCCESSFULLY!\n");
+    printf("NAPI_AsymmetricDecrypt json : %s\n", returnJsonChar);
+    plaintext_base64 = RetJsonObj::readData_string(returnJsonChar, "plaintext_base64");
+    printf("Decrypted plaintext : %s\n",base64_decode(plaintext_base64).c_str());
+    printf("NAPI_AsymmetricDecrypt data SUCCESSFULLY!\n");
```
Comment (yang8621) [inline core/App/ehsm_core_test.cpp:166]: Still need to compare the decrypted text and the plaintext.
Please also update the other tests.

File: ehsm_kms_service/test/test_kms_with_rest.py [hunk 6]
```diff
@@ -28,6 +28,38 @@ def test_params(payload):
...
+    params=test_params({
+            "cmk_base64":json.loads(create_resp.text)['result']['cmk_base64'],
+            "ciphertext_base64":json.loads(asymmetricEncrypt_resp.text)['result']['ciphertext_base64']
+        })
+    print('AsymmetricDecrypt req:\n%s\n' %(params))
+    asymmetricDecrypt_res = requests.post(url=base_url + "AsymmetricDecrypt", data=json.dumps(params), headers=headers)
+    print('AsymmetricDecrypt resp:\n%s\n' %(asymmetricDecrypt_res.text))
+    plaintext = str(base64.b64decode(json.loads(asymmetricDecrypt_res.text)['result']['plaintext_base64']), 'utf-8')
+    print('AsymmetricDecrypt plaintext:\n%s\n' %(plaintext))
+    print('====================test_RSA3072_encrypt_decrypt end===========================')
```
Comment (yang8621) [inline ehsm_kms_service/test/test_kms_with_rest.py:61]: compare the result with 123456 and print successfully then.
Update the other case as well, please.

File: ehsm_kms_service/test/test_kms_with_rest.py [hunk 7]
```diff
@@ -28,6 +28,38 @@ def test_params(payload):
...
+    create_resp = requests.post(url=base_url + "CreateKey", data=json.dumps(params), headers=headers)
+    print('CreateKey resp:\n%s\n' %(create_resp.text))
+
+    # test AsymmetricEncrypt("123456")
+    params=test_params({
+            "cmk_base64":json.loads(create_resp.text)['result']['cmk_base64'],
+            "plaintext":"123456"
+        })
+    print('AsymmetricEncrypt req:\n%s\n' %(params))
+    asymmetricEncrypt_resp = requests.post(url=base_url + "AsymmetricEncrypt", data=json.dumps(params), headers=headers)
```
Comment (yang8621) [inline ehsm_kms_service/test/test_kms_with_rest.py:48]: Check the return code on success or failure.

File: ehsm_kms_service/ehsm_kms_params.js (hunk 2 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/ehsm_kms_params.js:128]: It will change to 384.

File: ehsm_kms_service/ehsm_kms_params.js (hunk 1 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/ehsm_kms_params.js:117]: It will be fix

File: ehsm_kms_service/test/test_kms_with_rest.py (hunk 6 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/test/test_kms_with_rest.py:61]: It will be fix.

File: ehsm_kms_service/test/test_kms_with_rest.py (hunk 7 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/test/test_kms_with_rest.py:48]: It will be fix.

File: ehsm_kms_service/ehsm_kms_params.js (hunk 2 above)
Comment (syan10) [reply ehsm_kms_service/ehsm_kms_params.js:128]: for the maxlength in the nodejs,  previous patches also have the simlar issue, i remember that you told me that you want to refine the together?

File: ehsm_kms_service/test/test_kms_with_rest.py (hunk 6 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/test/test_kms_with_rest.py:61]: will be fix

File: ehsm_kms_service/ehsm_kms_params.js (hunk 1 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/ehsm_kms_params.js:117]: will be fix

File: ehsm_kms_service/test/test_kms_with_rest.py (hunk 7 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/test/test_kms_with_rest.py:48]: will be fix

File: ehsm_kms_service/ehsm_kms_server.js (hunk 4 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/ehsm_kms_server.js:179]: It will be processed in the JS layer

**Pull request closed without merging**